backend/src/test/java/com/redhat/cloud/notifications/events/LifecycleITest.java

```diff
@@ -412,8 +412,8 @@ private void pushMessage(int expectedWebhookCalls) throws IOException, Interrupt
             mockServerConfig.getMockServerClient().clear(expectedRequestPattern);
         }
 
+        counterAssertionHelper.awaitAndAssertIncrement(PROCESSED_MESSAGES_COUNTER_NAME, 1);
```

Comment:
That change was necessary because of a race condition in tests caused by the new SQL queries executed by `EventConsumer` (retrieve event type and create event). Without the `await` logic, the check was run before the Kafka message had been processed and failed.